torch/csrc/jit/script/compiler.cpp

```diff
@@ -372,16 +392,26 @@ at::optional<std::vector<Value*>> tryMatchSchema(
 
     std::vector<at::optional<NamedValue>> positional_inputs(schema.arguments.size(), at::nullopt);
 
-    size_t total_inputs = attributes.size() + inputs.size();
-    if(total_inputs > schema.arguments.size()) {
-      err() << "expected at most " << schema.arguments.size() << " arguments "
-      << " but found " << total_inputs << "\n" << loc << "\n";
-      return at::nullopt;
-    }
     // fill in positional arguments
-    for(size_t i = 0; i < inputs.size(); ++i) {
+    size_t non_vararg = std::min(inputs.size(), schema.arguments.size());
+    for(size_t i = 0; i < non_vararg; ++i) {
       positional_inputs[i] = inputs[i];
     }
+    // if we have extra vararg arguments pack them all up into a tuple as the last argument
+    // s.expand(3, 4) --> expand(s, (3, 4))
+    // s.expand(x) --> expand(s, x) # note: no tuple here it is possible that x is already a list
+    if(inputs.size() > schema.arguments.size()) {
+      if(!isVarargs(schema)) {
+        err() << "expected at most " << schema.arguments.size() << " positional arguments "
+        << " but found " << inputs.size() << "\n" << loc << "\n";
+        return at::nullopt;
+      }
+      auto last = inputs.back();
+      auto values = toValues(inputs.slice(schema.arguments.size() - 1));
```

Comment:
Shouldn't this be `.slice(1)` to remove the tensor only? Currently it will only leave the last int argument.